app/models/patient.rb

```diff
@@ -129,6 +129,136 @@ def latest_transfer
+  #   OR
+  #   patient is in continuous exposure
+  #   OR
+  #   patient last date of exposure is on or after (today - monitoring_period_days)
+  #   OR
+  #   patient last date of exposure is null and created at of exposure is on or after (today - monitoring_period_days)
+  scope :hoh_eligible_for_any_dependents, lambda {
+    joins(:dependents)
+      .where(purged: false)
+      .where(head_of_household: true)
+      .where('patients.id = patients.responder_id')
+      .where(
+        # Ignore any joined rows where the dependents_patients is the HoH itself.
+        'dependents_patients.id != dependents_patients.responder_id'
+      )
+      .has_usable_preferred_contact_method
+      .where(
+        # HoH is unconditionally ineligible if it has paused notifications
+        pause_notifications: false
+      )
+      .where('dependents_patients.monitoring = ?', true)
+      .where('dependents_patients.purged = ?', false)
+      .where(
+        'dependents_patients.isolation = ? '\
+        'OR dependents_patients.continuous_exposure = ? '\
+        'OR dependents_patients.last_date_of_exposure >= DATE_SUB(DATE(CONVERT_TZ(?, "UTC", dependents_patients.time_zone)), INTERVAL ? DAY) '\
+        'OR ('\
+        '  dependents_patients.last_date_of_exposure IS NULL '\
+        '  AND dependents_patients.created_at >= DATE_SUB(DATE(CONVERT_TZ(?, "UTC", dependents_patients.time_zone)), INTERVAL ? DAY)'\
+        ')',
+        true,
+        true,
+        Time.now.getlocal('-00:00'),
+        ADMIN_OPTIONS['monitoring_period_days'],
+        Time.now.getlocal('-00:00'),
+        ADMIN_OPTIONS['monitoring_period_days']
+      )
+      .within_preferred_contact_time
+  }
+
+  # Patients should be monitored are any of the below:
+  # - In isolation
+  # - In continuous exposure
+  # - last date of exposure is on or after (today - monitoring_period_days)
+  #   OR
+  #   last date of exposure is null and created at of exposure is on or after (today - monitoring_period_days)
+  scope :should_be_monitoring, lambda {
+    where(
+      'patients.isolation = ? '\
+      'OR patients.continuous_exposure = ? '\
+      'OR patients.last_date_of_exposure >= DATE_SUB(DATE(CONVERT_TZ(?, "UTC", patients.time_zone)), INTERVAL ? DAY) '\
+      'OR ('\
+      '  patients.last_date_of_exposure IS NULL '\
+      '  AND patients.created_at >= DATE_SUB(DATE(CONVERT_TZ(?, "UTC", patients.time_zone)), INTERVAL ? DAY)'\
+      ')',
+      true,
+      true,
+      Time.now.getlocal('-00:00'),
+      ADMIN_OPTIONS['monitoring_period_days'],
+      Time.now.getlocal('-00:00'),
+      ADMIN_OPTIONS['monitoring_period_days']
+    )
+  }
+
+  # A patient has not reported within the reporting period if either:
+  # - last assessment is null
+  # - lastest assessment date is before the beginning of the day in patient local
+  #   time for (time now - reporting_period_minutes).beginning of day
+  scope :has_not_reported_within_reporting_period, lambda {
```

Comment:
rename scope to be more descriptive of what it actually does. Do we need a scope that doesnt round to the day?

Reply:
renaming to `: has_not_reported_recently_enough_for_reminder_exlusion`